src/test/java/teammates/test/cases/common/FieldValidatorTest.java

```diff
@@ -521,65 +515,150 @@ public void testGetInvalidityInfoForEmail_invalid_returnErrorString() {
-        valueWithDisallowedChar = "cour@s*hy#";
-        testOnce("invalid: multiple disallowed chars",
-                FieldType.COURSE_ID,
-                valueWithDisallowedChar,
-                String.format(COURSE_ID_ERROR_MESSAGE, valueWithDisallowedChar, REASON_INCORRECT_FORMAT));
+        String maxLengthCourseId = StringHelper.generateStringOfLength(COURSE_ID_MAX_LENGTH);
+        assertEquals("Valid Course ID (max length) should return empty string", "",
+                     validator.getInvalidityInfoForCourseId(maxLengthCourseId));
     }
     
     @Test
+    public void testGetInvalidityInfoForCourseId_invalid_returnErrorString() {
+        String emptyCourseId = "";
+        assertEquals("Invalid Course ID (empty) should return appropriate error string",
+                     String.format(COURSE_ID_ERROR_MESSAGE, emptyCourseId, REASON_EMPTY),
+                     validator.getInvalidityInfoForCourseId(emptyCourseId));
+
+        String untrimmedCourseId = " $cs1101-sem1.2_ ";
+        assertEquals("Invalid Course ID (untrimmed) should return appropriate error string",
+                     String.format(WHITESPACE_ONLY_OR_EXTRA_WHITESPACE_ERROR_MESSAGE, COURSE_NAME_FIELD_NAME),
+                     validator.getInvalidityInfoForCourseId(untrimmedCourseId));
+
+        String whitespaceOnlyCourseId = "    ";
+        assertEquals("Invalid Course ID (whitespace only) should return appropriate error string",
+                     String.format(WHITESPACE_ONLY_OR_EXTRA_WHITESPACE_ERROR_MESSAGE, COURSE_NAME_FIELD_NAME),
+                     validator.getInvalidityInfoForCourseId(whitespaceOnlyCourseId));
+
+        String tooLongCourseId = StringHelper.generateStringOfLength(COURSE_ID_MAX_LENGTH + 1);
+        assertEquals("Invalid Course ID (too long) should return appropriate error string",
+                     String.format(COURSE_ID_ERROR_MESSAGE, tooLongCourseId, REASON_TOO_LONG),
+                     validator.getInvalidityInfoForCourseId(tooLongCourseId));
+
+        String courseIdWithSpaces = "my course id with spaces";
+        assertEquals("Invalid Course ID (contains spaces) should return appropriate error string",
+                     String.format(COURSE_ID_ERROR_MESSAGE, courseIdWithSpaces, REASON_INCORRECT_FORMAT),
+                     validator.getInvalidityInfoForCourseId(courseIdWithSpaces));
+
+        String courseIdWithInvalidChar = "cour@s*hy#";
+        assertEquals("Invalid Course ID (invalid char) should return appropriate error string",
+                     String.format(COURSE_ID_ERROR_MESSAGE, courseIdWithInvalidChar, REASON_INCORRECT_FORMAT),
+                     validator.getInvalidityInfoForCourseId(courseIdWithInvalidChar));
+    }
+
+    @Test
+    public void testGetInvalidityInfoForTimeForSessionStartAndEnd_valid_returnEmptyString() {
+        Date date = new Date();
+        Date sessionStart = new Date(date.getTime() - 1000);
+        Date sessionEnd = new Date(date.getTime() + 1000);
```

Comment:
One of our helper classes must/should have a utility method to get a time relative to current time. If not, add one?